src/commands/postDeploy/getLinuxDetectorError.ts

```diff
@@ -1,83 +1,95 @@
     const detectorUri: string = `${node.id}/detectors/${detectorId}`;
     const requestOptions: requestUtils.Request = await requestUtils.getDefaultAzureRequest(detectorUri, node.root);
-    const dayAfterDeployResultTime: Date = new Date(deployResultTime.getTime() + (24 * 60 * 60 * 1000));
 
-    // these parameters were specified to retrieve the timestamp from the detector by Detectors team
-    // string "Latest time seen by detector. To be used in VSCode integration."" is added to response
-    // when val: 'vscode' is added
     requestOptions.qs = {
         'api-version': "2015-08-01",
-        fId: "1",
-        btnId: "2",
-        val: "vscode",
-        inpId: "1",
-        startTime: deployResultTime.toISOString(),
-        endTime: dayAfterDeployResultTime.toISOString()
+        startTime,
+        endTime,
+        logFormat: 'plain'
     };
 
     const detectorResponse: string = await requestUtils.sendRequest<string>(requestOptions);
     const responseJson: detectorResponseJSON = <detectorResponseJSON>JSON.parse(detectorResponse);
 
-    const datasets: detectorDataset[] = responseJson.properties.dataset;
-    const vsCodeIntegration: string = 'Latest time seen by detector. To be used in VSCode integration.';
-    const timestampTable: detectorTable | undefined = findTableByRowValue(datasets, vsCodeIntegration);
+    const insightLogTable: detectorTable | undefined = findTableByName(responseJson.properties.dataset, 'insight/logs');
 
-    // if we can't find the timestamp, exit and try again
-    if (!timestampTable) {
+    if (!insightLogTable) {
         return undefined;
     }
 
-    const timestamp: Date = new Date(getValuesByColumnName(context, timestampTable, ColumnName.value));
-    if (validateTimestamp(context, timestamp, deployResultTime)) {
-
-        const criticalErrorTable: detectorTable | undefined = findTableByRowValue(datasets, 'Critical');
-        const failureCountTable: detectorTable | undefined = findTableByColumnName(datasets, ColumnName.failureCount);
-
-        // if the criticalError table or failureCounts aren't defined, we can't display anything to the user
-        if (!criticalErrorTable || !failureCountTable) {
+    const rawApplicationLog: string = getValuesByColumnName(context, insightLogTable, ColumnName.value);
+    const insightDataset: detectorDataset[] = <detectorDataset[]>JSON.parse(rawApplicationLog);
+
+    let insightTable: detectorTable;
+    let detectorTimestamp: RegExpMatchArray | null;
+
+    const appInsightTable: detectorTable | undefined = findTableByName(insightDataset, 'application/insight');
+    if (appInsightTable) {
+        insightTable = appInsightTable;
+        context.telemetry.properties.insight = 'app';
+        detectorTimestamp = getValuesByColumnName(context, appInsightTable, ColumnName.dataName).match(timestampFormat);
```

Comment:
Can you give me example data of what's going on here? I'm a little confused what the regexp is doing and why it's necessary

Reply:
![image](https://user-images.githubusercontent.com/5290572/79900270-d9d91c80-83c2-11ea-9460-611712bc34c2.png)

I'm looking for the timestamp from the insight and comparing it to the deployEndTime to make sure that we're not looking at stale data.

Reply:
Why does it have "[1] " in front of the time? I'm just worried about the format of this changing in the future because it seems really weird. Ideally the RegExp we use is super simple and mostly just passes along the time and lets `new Date` parse it in whatever format it comes in

Reply:
> I added the [1] because the insight table is internally Dictionary<string,object> and the key string needs to be unique. So if there are multiple issues happened at the same timestamp, then it will throw an exception.

Reply:
That makes very little sense to me. What's the key and what's the value? Why would a timestamp be a key?

That aside, instead of a regexp for the time part, seems like you could have a simpler regexp to cut out the `[1] ` part